fem/fe.hpp

```diff
@@ -228,18 +228,18 @@ class KnotVector;
 // Base and derived classes for finite elements
 
 
-/// Abstract class for Finite Elements
+/// Abstract class for all finite elements.
 class FiniteElement
 {
 protected:
-   int Dim;      ///< Dimension of reference space
-   Geometry::Type GeomType; ///< Geometry::Type of the reference element
-   int FuncSpace, RangeType, MapType,
-       DerivType, DerivRangeType, DerivMapType;
+   int dim;      ///< Dimension of reference space
```

Comment:
Yeah I went down this road when Mark and others didn't like my RangeT compromise.  The problem before was that the enums were unname so you couldn't have references to them in Doxygen, so I had to pick something an d nobody liked what I picked.  I think the testing should catch most adverse effects here since this is all so foundational, but one can never be sure.  @tzanio @mlstowell what do you think?

Reply:
Sorry, I know this is a hassle.  Having named enumerations is definitely useful for Doxgen.  However, the reason I like named enumerations is that they allow the compiler to catch silly mistakes that I sometimes make.  Not all compilers type check enums but some do.  This PR won't really help much with type safety, that's not the goal here, but it is a step in that direction.  One day I'd like to see these member data stored using the enum type names rather than `int` and change all the accessor methods to return the appropriate enum type.  It's not a high priority but it could help those with OCD.